aces-nifi-processors/src/main/java/net/acesinc/nifi/processors/mongodb/PartialUpdateMongo.java

```diff
@@ -353,22 +414,80 @@ protected void transferFlowFile(FlowFile f, Object result, ProcessContext contex
+                        logger.debug("single propertyName scenario w/ parent.child notation: $addToSet, jsonString:=" + jsonString);
+                        Map<String, Object> operationValueDocs = mapper.readValue(jsonString, Map.class);
+                        Document parentDoc = this.fixChildOfParentForOpAddToSetEachArray(parentPropertyName, childPropertyName, operationValueDocs);
+                        if (!parentDoc.isEmpty()) {
+                            updateDocument.append(operation, parentDoc);
+                        } else {
+                            logger.error("Empty document returned when attempting $addToSet-$each w/ parent.child w/ single propertyName scenario. Scenario: $addToSet, propertyName [" + propertyName + "], parentPropertyName [" + parentPropertyName + "], childPropertyName [" + childPropertyName + "]");
+                        }
+                    }
+                }
+            } else if (OPERATION_SET.equals(operation) && propertyName.contains(".")) {
+                //logic on when $set handles an object -- children -- parent.child item specified
+                String[] tmps = propertyName.split("\\.");
+                if (tmps.length != 2) {
+                    throw new IllegalStateException("IllegalState thrown on $set: Use of '.' denotes parent child relationship and is allowed one child dimension in this processor. propertyName [" + propertyName + "], length [" + tmps.length + "]");
+                }
+                String parentPropertyName = tmps[0];
+                String childPropertyName = tmps[1];
+                if (doc.get(parentPropertyName) != null && mapper.writeValueAsString(doc.get(parentPropertyName)).contains("{")) {
+                    String jsonString = mapper.writeValueAsString(doc.get(parentPropertyName));
+                    logger.debug("single propertyName scenario w/ parent.child notation: $set, propertyName [" + propertyName + "], parentPropertyName [" + parentPropertyName + "], childPropertyName [" + childPropertyName + "]");
+                    logger.debug("single propertyName scenario w/ parent.child notation: $set, jsonString:=" + jsonString);
+                    Map<String, Object> operationValueDocs = mapper.readValue(jsonString, Map.class);
+                    operationValueDocs = this.fixChildOfParentForOpSet(parentPropertyName, childPropertyName, operationValueDocs);
+                    if (!operationValueDocs.isEmpty()) {
+                        updateDocument.append(operation, operationValueDocs);
+                    } else {
+                        logger.error("Empty document returned when attempting $set w/ parent.child w/ single propertyName scenario. Scenario: $set, propertyName [" + propertyName + "], parentPropertyName [" + parentPropertyName + "], childPropertyName [" + childPropertyName + "]");
+                    }
```

Comment:
This code looks like it's the same as the above `if`. Move into a method for reuse?

Reply:
Ok, there is a slight difference when we are dealing w/ multiple propertyNames vs one propertyName, especially at the very end of the process for each of those scenarios after we massage what we need to be in the Document.  But, there is some repetition.  However, I do need those debug statements the way they are regardless of what I do.

Reply:
Just wanted to repeat myself in another place, debug is turned off by default on the logback.xml on nifi, so we won't see these debugs on production.  But, any debugs that remain that are for the parent.child scenarios need to remain to me.